### PR TITLE
ChangeFeedProcessor: Adds AllVersionsAndDeletes support to ChangeFeedProcessor

### PR DESCRIPTION
**Pull request by philipthomas-MSFT**
# Pull Request Template

## Description

Adding **AllVersionsAndDeletes** to `ChangeFeedProcessor`.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 1]
```diff
@@ -1681,6 +1680,81 @@ public abstract class Container
...
+        /// await this.Container.DeleteItemAsync<dynamic>(id: "1", partitionKey: new PartitionKey("1"));
+        /// 
+        /// allProcessedDocumentsEvent.WaitOne(10 * 1000);
+        /// 
+        /// await changeFeedProcessor.StopAsync();
+        /// ]]>
+        /// </code>
+        /// </example>
+        /// <returns>An instance of <see cref="ChangeFeedProcessorBuilder"/></returns>
+        public abstract ChangeFeedProcessorBuilder GetChangeFeedProcessorBuilderWithAllVersionsAndDeletes<T>(
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:1755]: You will need to run UpdateContracts.ps1

Comment (philipthomas-MSFT) [reply Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:1755]: Ah. Thanks for that reminder.

Comment (philipthomas-MSFT) [reply Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:1755]: Done

Comment (FabianMeiswinkel) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:1755]: Is this name intentional? Why not have a parameter with ChangeFeedMode?

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:1755]: I think this was already discussed. It was the way to, from the API perspective, make people use the metadata model (ChangeFeedItem). Whether or not this name is too long we can certainly discuss and change.

Comment (philipthomas-MSFT) [reply Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:1755]: Fine with the name and approach staying this way. Both naming and approach is more intentional. Length of name should not be a problem as long as its intent is understandable. And `GetChangeFeedProcessorBuilderWithManualCheckpoint` is another long one, but it makes sense, and we know its intent.

Comment (philipthomas-MSFT) [reply Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:1755]: Can we resolve this? Or do we need to have an offline conversation? Thanks.

Comment (FabianMeiswinkel) [reply Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:1755]: If it has been discussed - I am fine resolving it. Don't like it - but it is fine to use different approaches when it comes to naming I guess.,

**Review by FabianMeiswinkel — DISMISSED** [review]
Given the naming has been discussed before and was agreed-upon - LGTM

**Comment by philipthomas-MSFT** [comment]
> Given the naming has been discussed before and was agreed-upon - LGTM

Much appreciated Fabian. I will make a note to bring other SDK and Principals in on these discussions if necessary moving forward.

**Review by ealsur — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs (hunk 1 above)
Comment (jcocchi) [reply Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:1755]: One note about naming, I'd like to see `ChangeFeedItemChange` simplified to `ChangeFeedItem`, but we agreed to handle that in a separate PR

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

**Pull request merged**